### ISSUE TITLE
Accuracy or similar metrics used for `autonet.score`

### ISSUE THREAD

**Issue by maxmarketit** [automl/Auto-PyTorch#55]
I found that `autonet.score` is different from the loss for training.

  As I am using custom loss function, it would be more convenient and appropriate to have

  custom score function. Anyway thank you for the package.

**Comment by LMZimmer** [comment]
Hey,
there are three different scorings:

i) Training loss (e.g. `cross_entropy`) that is used for the optimizer when training an NN

ii) Metrics for scoring (e.g. `accuracy_score`), these are logged in results.json under their name

iii) The loss considered by BOHB. This is the main metric (could be `accuracy_score`) transformed to a metric that is minimizable, that is `1-accuracy_score`. This loss is logged as `loss`.

You can change ii) via the `optimize_metric` argument in the autonet config.

**Comment by maxmarketit** [comment]
I wrote a code with new optimization metric

```
def qloss(y_true, y_pred, qs):
        e = y_true - y_pred
        return np.maximum(qs * e, (qs-1)*e).mean()
metric_ql = functools.partial(qloss, qs = np.array([[0.1, 0.5, 0.9]]))

def qinputloss(y_true, y_pred, ws):
    w_q = ws[0,0]
    w_input = ws[0:1, 1:]
    q = y_true[:, 0:1]

    e_q = np.abs(y_true[:,0]-y_pred[:,0])
    e = y_true[:, 1:] - y_pred[:, 1:]

    #print(w_input)
    #print(q*e, (q-1)*e)

    ret = w_input * np.maximum(q*e, (q-1)*e)
    #print(ret)

    return np.mean(ret)+ np.mean(w_q*e_q)
metric_qil = functools.partial(qinputloss, ws=np.array([[0,1]]))

metric_selector = autonet.pipeline[MetricSelector.get_name()]
metric_selector.add_metric('metric_ql', metric_ql, loss_transform=False, requires_target_class_labels=False)
metric_selector.add_metric('metric_qil', metric_qil, loss_transform=False, requires_target_class_labels=False)

results_fit = autonet.fit(X_train=train_x_wq,
                          Y_train=train_y_wq,
                          
                          loss_modules=["qinput_loss"], 
                          optimize_metric = 'metric_qil')
						  
```

I have not test it fully, but it looks working

**Comment by LMZimmer** [comment]
Looking at your code, it looks like you are adding your metrics to the metric selector (which is used to choose configurations with BOHB) but also add `loss_modules` (which is the module used as training loss).

If you want to just use the metric, you should omit selecting the loss module (which will use cross_entropy per default). If you actually want to use a different training loss, you have to add it to the respective [node](https://github.com/automl/Auto-PyTorch/blob/a93c1e0d15442c19a0c5825b79c48f4b33ad8943/autoPyTorch/pipeline/nodes/loss_module_selector.py#L42).
